### PR TITLE
Land type lists and land type tags

### PR DESCRIPTION
**Pull request by kirderf1**
Does two things:
- Replaces land type groups with tags
- Which land types that can get picked at random is now declared through json
This decouples the land types themselves from how they are chosen, as well as making it easier to get an overview of which lands are available
The json aspect also makes it easier to tweak these things

While not part of this PR, it should be a bit easier to introduce weights in the future if we ever want to.
This is not a particularly important PR, and could be held off until after the 4/13 update if people want to discuss the directionof the PR first. I'll hold off on adding to the changelog until I know whether it will go in the 4/13 update or not.

### CODE REVIEW

**Comment by kirderf1** [comment]
Ah, and there are some minor code cleanup in association with the switch from a group name to using tags

File: src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+					List<GroupData> parsedData = TERRAIN_DATA_CODEC.parse(JsonOps.INSTANCE, json).getOrThrow(false, LOGGER::error);
+					terrainData.addAll(parsedData);
+				} catch(IOException ignored)
+				{
+				} catch(RuntimeException runtimeexception)
+				{
+					LOGGER.warn("Invalid json in data pack: '{}'", location.toString(), runtimeexception);
+				}
+			});
+		}
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java:119]: Could you put a comment at the beginning with a brief description of what happens in the for loop

File: src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+					for(EnumAspect aspect : parsedData.keySet())
+						titleData.get(aspect).addAll(parsedData.get(aspect));
+					
+				} catch(IOException ignored)
+				{
+				} catch(RuntimeException runtimeexception)
+				{
+					LOGGER.warn("Invalid json in data pack: '{}'", location.toString(), runtimeexception);
+				}
+			});
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java:142]: Could you put a comment at the beginning with a brief description of what happens in the for loop

File: src/main/java/com/mraof/minestuck/item/loot/conditions/LandTypeLootCondition.java [hunk 3]
```diff
@@ -52,7 +59,7 @@ public boolean test(LootContext context)
 		LandTypePair aspects = LandTypePair.getTypes(level).orElse(null);
 		
 		if(aspects != null && (terrainTypes.contains(aspects.getTerrain()) || titleTypes.contains(aspects.getTitle())
-				|| terrainGroups.contains(aspects.getTerrain().getGroup()) || titleGroups.contains(aspects.getTitle().getGroup())))
+				|| terrainTag != null && aspects.getTerrain().is(terrainTag) || titleTag != null && aspects.getTitle().is(titleTag)))
```
Comment (Dweblenod) [inline src/main/java/com/mraof/minestuck/item/loot/conditions/LandTypeLootCondition.java:62]: Feel free to ignore but we should probably split this into a couple booleans or something for easier reading. We have so many OR operators here!

**Review by Dweblenod — COMMENTED** [review]
Made a few suggestions to improve readability, but I have no complaints/concerns in terms of what this brings to the mod! Let me know if you want me to test it out in game, otherwise feel free to merge at your own discretion once considering the before mentioned suggestions

File: src/main/java/com/mraof/minestuck/item/loot/conditions/LandTypeLootCondition.java (hunk 3 above)
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/item/loot/conditions/LandTypeLootCondition.java]: Added some newlines to improve readability. Does that work?

File: src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java (hunk 2 above)
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java]: Instead split it into functions that should have the same effect.

File: src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java (hunk 1 above)
Comment (kirderf1) [reply src/main/java/com/mraof/minestuck/world/lands/gen/LandTypeSelectionLoader.java]: Instead split it into functions that should have the same effect.

File: src/main/java/com/mraof/minestuck/item/loot/conditions/LandTypeLootCondition.java (hunk 3 above)
Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/item/loot/conditions/LandTypeLootCondition.java]: yup!

**Review by Dweblenod — APPROVED** [review]
Once info is entered into the Changelog, and as long as its been tested, Im all good for it to go!

**Pull request merged**